### PR TITLE
add jgi-mg-interaface sample data files

### PR DESCRIPTION
**Pull request by brynnz22**
For issue #114 : @turbomam @mslarae13  I added the jgi-mg-interface sample data files. There was one unexpected pass for the `dna_sample_name` slot. I was able to add a colon in it when it should only allow "a-z, A-Z, 0-9, - and _".

### CODE REVIEW

**Comment by turbomam** [comment]
> ...`dna_sample_name` ...  I was able to add a colon in it when it should only allow "a-z, A-Z, 0-9, - and _".

Thanks for the PR, @brynnz22 . I'm going to merge it now.

Can you say how you determined that `dna_sample_name` should only allow a-z, A-Z, 0-9, - and _? Is that in one of our schemas and I just missed it?! Or are you referring to a specification from some other source, like JGI? If that's the case, could you share the JGI specification, preferably in a computable format?

File: src/data/unexpected_pass/SampleData-jgi_mg_data-colon-dna_sample_name.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    dna_concentration: 1.23
+    dna_cont_type: plate
+    dna_cont_well: C3
+    dna_container_id: xxx
+    dna_dnase: "no"
+    dna_isolate_meth: xxx
+    dna_project_contact: xxx
+    dna_samp_id: xxx
+    dna_sample_format: DNAStable
+    dna_sample_name: DNA:0546789 # contains colon and should not be allowed 'a-z, A-Z, 0-9, - and _ only'
```
Comment (turbomam) [inline src/data/unexpected_pass/SampleData-jgi_mg_data-colon-dna_sample_name.yaml:14]: Thanks. I am going to merge this, but I would like to hear more about how your legal character list was determined.

**Review by turbomam — APPROVED** [review]

File: src/data/unexpected_pass/SampleData-jgi_mg_data-colon-dna_sample_name.yaml (hunk 1 above)
Comment (brynnz22) [reply src/data/unexpected_pass/SampleData-jgi_mg_data-colon-dna_sample_name.yaml:14]: Thanks Mark! For the dna_sample_name, the submission schema documentation - under the LinkML Source section mentions this. It says: "Sample names must be unique across all JGI projects and contain a-z, A-Z, 0-9, - and _ only."

https://microbiomedata.github.io/submission-schema/dna_sample_name/

Comment (turbomam) [reply src/data/unexpected_pass/SampleData-jgi_mg_data-colon-dna_sample_name.yaml:14]: OK, I guess we never implemented that!

**Pull request merged**